utilities/xmlrpcpp/src/XmlRpcDispatch.cpp

```diff
@@ -81,84 +86,69 @@ XmlRpcDispatch::work(double timeout)
   _endTime = (timeout < 0.0) ? -1.0 : (getTime() + timeout);
   _doClear = false;
   _inWork = true;
+  int timeout_ms = static_cast<int>(floor(timeout * 1000.));
 
   // Only work while there is something to monitor
   while (_sources.size() > 0) {
 
     // Construct the sets of descriptors we are interested in
-    fd_set inFd, outFd, excFd;
-	  FD_ZERO(&inFd);
-	  FD_ZERO(&outFd);
-	  FD_ZERO(&excFd);
+    struct pollfd fds[_sources.size()];
 
-    int maxFd = -1;     // Not used on windows
     SourceList::iterator it;
-    for (it=_sources.begin(); it!=_sources.end(); ++it) {
+    std::size_t i = 0;
```

Comment:
I would prefer the scope of this variable to be limited to the for loop since it isn't used afterwards.

The loop in line 119 can then declare its own local index.